### PR TITLE
Fix #275 - Add link on Monitoring Feeds page to view most recent feed response

### LINKED ISSUE #275: Add link on Monitoring Feeds page to view most recent feed response
**Summary:** 

Currently, if you run the validator tool on a feed and it has no errors or warnings, there is no way to view the most recent feed response in plain text (see below).  

image

The Iteration Details page is currently only visible if an error or warning occurs, and then you can click on the iteration number in the Log/Summary.

We should add a link that you could click on for each feed that would take you to the latest feed iteration in the Iteration Details view - here is a mockup of a hyperlink right below the HTTP statistics:

image

If there are errors or warnings they would be shown like normal on the right, if there weren't any errors/warnings then the right panel would be empty (below is the normal view when there are errors/warnings).

image

**Steps to reproduce:** 

Run the tool on a feed without any errors/warnings

**Expected behavior:** 

Give me a way to view the most recent feed message retrieved from the server

**Observed behavior:** 

There is no way to view the most recent GTFS-rt feed message.

**Platform:** 

N/A

### PR DESCRIPTION
**Pull request by Suryakandukoori**
**Summary:**

Now a link to check the feed message available in the error summary area of the feed monitoring page.

**Expected behavior:** 

The link will open an iteration details screen which shows the latest feed message if there are no errors and the latest feed message along with errors if any.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [X] Run the unit tests with `mvn test` to make sure you didn't break anything
- [X] Format the title like "Fix #issue - short description of fix and changes"
- [X] Linked all relevant issues
- [X] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: src/main/java/edu/usf/cutr/gtfsrtvalidator/api/resource/GtfsRtFeed.java [hunk 1]
```diff
@@ -337,17 +345,25 @@ public Response getIterationErrors(
...
+            @PathParam("iterationId") int iterationId,
+            @PathParam("gtfsRtId") int gtfsRtId) {
 
         GtfsRtFeedIterationModel gtfsRtFeedIterationModel;
         Session session = GTFSDB.initSessionBeginTrans();
 
-        gtfsRtFeedIterationModel = (GtfsRtFeedIterationModel) session.createQuery(" FROM GtfsRtFeedIterationModel" +
-                                                                                      " WHERE IterationId = " + iterationId)
-                                                                                      .uniqueResult();
+        if(iterationId > -1) {
```
Comment (barbeau) [inline src/main/java/edu/usf/cutr/gtfsrtvalidator/api/resource/GtfsRtFeed.java]: Needs a space between `if` and `(`.

File: src/main/java/edu/usf/cutr/gtfsrtvalidator/api/resource/GtfsRtFeed.java [hunk 2]
```diff
@@ -259,16 +259,24 @@ public Response getMonitorData(
 
     // Returns feed message for a requested iteration
      @GET
-     @Path("/{iterationId : \\d+}/feedMessage")
+     @Path("/{iterationId : -?\\d+}/{gtfsRtId : \\d+}/feedMessage")
      @Produces(MediaType.APPLICATION_JSON)
      public String getFeedMessage(
-             @PathParam("iterationId") int iterationId) {
+             @PathParam("iterationId") int iterationId,
```
Comment (barbeau) [inline src/main/java/edu/usf/cutr/gtfsrtvalidator/api/resource/GtfsRtFeed.java]: I think these would both make more sense as `@QueryParam`?  Let me know what you think.

File: src/main/resources/webroot/custom-js/iteration.js [hunk 3]
```diff
@@ -29,9 +30,12 @@ var allTablesData = '';
 
 var server = window.location.protocol + "//" + window.location.host;
 $(".feed-page-loader").show();
-$(".issues-page-loader").show();
-
-$.get(server + "/api/gtfs-rt-feed/" + iterationId + "/iterationDetails").done(function (data) {
+if(iterationId != -1) {
```
Comment (barbeau) [inline src/main/resources/webroot/custom-js/iteration.js]: Space between `if` and `(`.

**Comment by Suryakandukoori** [comment]
@barbeau this branch somehow does not have all my changes. I will redo some commits and refresh this pull request.

File: src/main/java/edu/usf/cutr/gtfsrtvalidator/api/resource/GtfsRtFeed.java (hunk 2 above)
Comment (Suryakandukoori) [reply src/main/java/edu/usf/cutr/gtfsrtvalidator/api/resource/GtfsRtFeed.java]: I made the related changes in the review API parameters issue #216

Comment (Suryakandukoori) [reply src/main/java/edu/usf/cutr/gtfsrtvalidator/api/resource/GtfsRtFeed.java]: I am going to raise a pull request for #216 as soon as we merge this request

Comment (barbeau) [reply src/main/java/edu/usf/cutr/gtfsrtvalidator/api/resource/GtfsRtFeed.java]: Sounds good, we'll change the parameters in a separate pull request.

**Comment by barbeau** [comment]
Thanks, looks good!

**Pull request merged**